### ISSUE TITLE
Autocomplete for qualified values not optimal

### ISSUE THREAD

**Issue by thought2** [nwolverson/vscode-ide-purescript#218]
Say I have multiple modules containing a value `view`.

No I'd like to import one specific in a qualified way:

When I write
`Foo.view`
and trigger AutoCompletion, I'm getting suggestions for all modules exporting `view`, but not the one from say `Something.Foo` listed at the top. So it is quite some manual work to scroll to the list and find the one I need.

Is there something we can do about this on the level of this extension?

**Comment by nwolverson** [comment]
There is currently a bunch of scoring done to order these completions, but mostly around the name of the prefix & module chosen, so `Something.Foo` should actually appear very near the top of the list if you complete `Foo.view` (because `Foo` matches a component of the module name).

If I gather correctly you're suggesting sorting modules that are already imported unqualified (e.g. `import Something.Foo (bar)`) higher? That seems reasonable but I couldn't say for sure if it's feasible without checking.

**Comment by thought2** [comment]
Oh wow, now I tried this again and realized that it works exactly how you say for my simplified example.
In my case I was importing a value 'view' from module `App.UI.Slider` but I wanted it qaulified as `UI.Slider.view` which would result in the import
`import App.UI.Slider as UI.Slider`

And in this case if i write "UI.Slider.view" (nothing is imported at this point) I am not seeing `App.UI.Slider` listed near the top.

This way of importing qualified as a name that contains a dot itself was maybe not considered in the matching function? If you  point me to the responsible code section, I can see if I can provide a fix.

**Comment by thought2** [comment]
But for now I will just import as UISlider and I'm good :)

**Comment by nwolverson** [comment]
Right, I have come across similar cases where we could do better relating the qualifier.

The suggestion ranking is around here in the language server: https://github.com/nwolverson/purescript-language-server/blob/main/src/LanguageServer/IdePurescript/Completion.purs#L200
